### PR TITLE
Add linker to area owners

### PR DESCRIPTION
**Pull request by agocke**
Add a new area-tools-linker and assign myself as the lead and @sbomer as technical expert

### CODE REVIEW

**Review by sbomer — APPROVED** [review]
LGTM. I wonder if @vitek-karas wants to be added to the list.

File: .github/fabricbot.json [hunk 1]
```diff
@@ -173,6 +173,25 @@
             "vsadov"
           ]
         },
+        {
+          "labels": [
+            "area-tools-linker"
```
Comment (jkotas) [inline .github/fabricbot.json:178]: So is it going to be `ILLink` or `linker` ?

Comment (jkotas) [reply .github/fabricbot.json:178]: https://github.com/dotnet/runtime/pull/77569#discussion_r1014525023

Comment (agocke) [reply .github/fabricbot.json:178]: I'm fine with ILLink. I'll change the label name

**Comment by agocke** [comment]
Added Vitek as well

**Comment by danmoseley** [comment]
You'll presumably want to label whatever you can (transferred issues and any PRs) and then ask Drew to retrain the labeler.

File: docs/area-owners.md [hunk 2]
```diff
@@ -134,6 +134,7 @@ Note: Editing this file doesn't update the mapping used by `@msftbot` for area-s
...
 | area-TieredCompilation-coreclr                 | @mangod9      | @kouvel                                             |                                                                                                                                                                                                                                                                                           |
+| area-Tools-ILLink                | @agocke      | @sbomer                                             |                                                                                                                                                                                                                                                                                           |
```
Comment (marek-safar) [inline docs/area-owners.md:137]: Could you introduce @dotnet/area-illink alias, it would help to tag all interested folks ad hoc as well

**Review by jeffschwMSFT — APPROVED** [review]
lgtm

**Comment by joperezr** [comment]
@agocke did we validate that the change was valid in the fabricbot portal? I'm seeing that fabricbot automations are not running today, and when I try to load the current configuration in fabricbot for this repo it fails to parse and says that there are no bot configurations tasks for this repository.

**Comment by joperezr** [comment]
FYI: @jeffhandley @ericstj @eiriktsarpalis

**Pull request merged**